pkg/assets/store.go

```diff
@@ -389,6 +389,64 @@ func (cos *cacheOnlyStore) TLSAsset(sel interface{}) string {
 	return k.toString()
 }
 
+// HasObject checks if the object exists in the underlying store.
+// This method is only used by external clients of the assets package such as the OpenTelemetry collector operator.
+func (s *StoreBuilder) HasObject(obj interface{}) (bool, error) {
+	if obj == nil {
+		return false, errors.New("object cannot be nil")
+	}
+
+	key, err := assetKeyFunc(obj)
+	if err != nil {
+		return false, fmt.Errorf("failed to get key for object: %w", err)
+	}
+
+	_, exists, err := s.objStore.GetByKey(key)
+	if err != nil {
+		return false, fmt.Errorf("failed to check object in store: %w", err)
+	}
+
+	return exists, nil
+}
+
+// AddObject adds an object to the underlying store.
+// This method is only used by external clients of the assets package such as the OpenTelemetry collector operator.
+func (s *StoreBuilder) AddObject(obj interface{}) error {
+	if obj == nil {
+		return errors.New("object cannot be nil")
+	}
+
+	if err := s.objStore.Add(obj); err != nil {
+		return fmt.Errorf("failed to add object to store: %w", err)
+	}
+
+	return nil
+}
+
+// GetObject retrieves an object from the underlying store.
+// This method is only used by external clients of the assets package such as the OpenTelemetry collector operator.
+func (s *StoreBuilder) GetObject(obj interface{}) (interface{}, error) {
```

Comment:
I'd go with a bool return value that returns whether the object is found or not to avoid the HasObject() method.
```suggestion
func (s *StoreBuilder) GetObject(obj interface{}) (interface{}, bool, error) {
```

Reply:
Thanks @simonpasquier. I was doing some integration testing with the same changes, hence had converted my PR to draft, yes GetObject serves the purpose and we don't need an additional method. Have removed it.